test/Microsoft.NET.Build.Tests/GivenThatWeWantToVerifyNuGetReferenceCompat.cs

```diff
@@ -0,0 +1,227 @@
+        }
+
+        public void Dispose()
+        {
+            DeleteNuGetArtifacts();
+        }
+
+        private void DeleteNuGetArtifacts()
+        {
+            try
+            {
+                //  Delete the shared NuGet package directory before running all the tests.
+                string directory = Path.Combine(RepoInfo.GetBaseDirectory(), ConstantStringValues.IdentifierDirectoryPrefix + ConstantStringValues.NuGetSharedDirectoryIdentifierPostfix);
+                if (Directory.Exists(directory))
+                {
+                    Directory.Delete(directory, true);
+                }
+                //  Delete the generated NuGet packages in the cache.
+                foreach (string dir in Directory.EnumerateDirectories(RepoInfo.NuGetCachePath, ConstantStringValues.DependencyPrefix + "*"))
+                {
+                    Directory.Delete(dir, true);
+                }
+            }
+            catch
+            {
+                // No-Op; as this is a precaution - do not throw an exception.
+            }
+        }
+    }
+
+    [CollectionDefinition("GivenThatWeWantToVerifyNuGetReferenceCompatFixture Collection")]
+    public class GivenThatWeWantToVerifyNuGetReferenceCompatFixtureCollection : ICollectionFixture<GivenThatWeWantToVerifyNuGetReferenceCompatFixture>
+    { }
+
+    [Collection("GivenThatWeWantToVerifyNuGetReferenceCompatFixture Collection")]
+    public class GivenThatWeWantToVerifyNuGetReferenceCompat : SdkTest
+    {
+        GivenThatWeWantToVerifyNuGetReferenceCompatFixture fixture;
+
+        public GivenThatWeWantToVerifyNuGetReferenceCompat(GivenThatWeWantToVerifyNuGetReferenceCompatFixture fixture)
+        {
+            this.fixture = fixture;
+        }
+
+        [Theory]
+        [InlineData("net45", "Full", "netstandard1.0 netstandard1.1 net45", true, true)]
+        [InlineData("net451", "Full", "netstandard1.0 netstandard1.1 netstandard1.2 net45 net451", true, true)]
+        [InlineData("net46", "Full", "netstandard1.0 netstandard1.1 netstandard1.2 netstandard1.3 net45 net451 net46", true, true)]
+        [InlineData("net461", "PartM3", "netstandard1.0 netstandard1.1 netstandard1.2 netstandard1.3 netstandard1.4 net45 net451 net46 net461", true, true)]
+        [InlineData("net462", "PartM2", "netstandard1.0 netstandard1.1 netstandard1.2 netstandard1.3 netstandard1.4 netstandard1.5 net45 net451 net46 net461", true, true)]
+        //  Fullframework NuGet versioning on Jenkins infrastructure issue
+        //        https://github.com/dotnet/sdk/issues/1041
+        //[InlineData("net461", "Full", "netstandard1.0 netstandard1.1 netstandard1.2 netstandard1.3 netstandard1.4 netstandard1.5 netstandard1.6 netstandard2.0 net45 net451 net46 net461", true, true)]
+        //[InlineData("net462", "Full", "netstandard1.0 netstandard1.1 netstandard1.2 netstandard1.3 netstandard1.4 netstandard1.5 netstandard1.6 netstandard2.0 net45 net451 net46 net461", true, true)]
```

Comment:
I would prefer not to have to unravel the logic; the follow-up action after the resolution of #1041 seems simpler here.